src/cleanvision/dataset/dataset.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+from collections.abc import Sized
+from typing import TYPE_CHECKING, List, Optional, Union, Dict
+
+from PIL import Image
+import pandas as pd
+from cleanvision.utils.utils import get_filepaths
+
+if TYPE_CHECKING:  # pragma: no cover
+    import datasets
+    from torchvision.datasets.vision import VisionDataset
+
+
+class Dataset(Sized):
```

Comment:
why is `cleanlab.dataset` a module if it only contains this single file?

Did you mean to instead add the HFDataset and TorchDataset in different files within the same module?  That sounds reasonable to me, or just not making this a module and keeping it as a file instead.

Reply:
I didn't find it right to put it with utils or issue_managers and the different classes in it are not big enough so didn't move it to separate files.
I can refactor to
```
src
- dataset
    - __init__.py (contains base class)
    - folder_dataset.py
    - hf_dataset.py

```and so on

Reply:
No worries I think it's fine to just keep it as is!